### PR TITLE
Add benchmark to validate variable-length edge filters

### PR DESCRIPTION
**Pull request by jeffreylovitz**
Introduce a new benchmark scenario to validate the enhancement in #1657.

### CODE REVIEW

File: tests/benchmarks/variable_length_filter.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    - connections: 32
+    - requests: 1000000
+    - queries:
+      - { q: "MATCH (a)-[e*]->(b) WHERE e.connects = \"CD\" RETURN a.name, b.name", ratio: 0.25 }
+      - { q: "MATCH (a)-[e*]->(b) WHERE e.connects = \"CD\" OR e.connects = \"DE\" RETURN a.name, b.name", ratio: 0.25 }
+      - { q: "MATCH (a)-[e*]->(b) WHERE e.connects = \"CD\" OR e.connects = \"DE\" OR e.connects = \"EF\" RETURN a.name, b.name", ratio: 0.25 }
+      - { q: "MATCH (a)-[e*]->(b) WHERE e.connects = \"CD\" OR e.connects = \"DE\" OR e.connects = \"EF\" OR e.connects = \"FG\" RETURN a.name, b.name", ratio: 0.25 }
+kpis:
+  - le: { $.OverallClientLatencies.Total.q50: 1.0 }
+  - ge: { $.OverallQueryRates.Total: 25000 }
```
Comment (swilly22) [inline tests/benchmarks/variable_length_filter.yml:23]: remind me what's the meaning of `OverallQueryRates.Total` ?

File: tests/benchmarks/variable_length_filter.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+remote:
+  - setup: redisgraph-r5
+  - type: oss-standalone
+dbconfig:
+  - dataset: "datasets/highly_connected.rdb"
+clientconfig:
+  - tool: redisgraph-benchmark-go
+  - parameters:
+    - graph: "g"
+    - rps: 0
```
Comment (swilly22) [inline tests/benchmarks/variable_length_filter.yml:11]: `0` stands for unlimited?

File: tests/benchmarks/variable_length_filter.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+name: "VARIABLE-LENGTH-FILTER"
+remote:
+  - setup: redisgraph-r5
+  - type: oss-standalone
```
Comment (swilly22) [inline tests/benchmarks/variable_length_filter.yml:4]: Can this be the default ?

**Review by swilly22 — APPROVED** [review]

File: tests/benchmarks/variable_length_filter.yml (hunk 2 above)
Comment (filipecosta90) [reply tests/benchmarks/variable_length_filter.yml:11]: yes :+1: . Not specifying rps also has the same effect

File: tests/benchmarks/variable_length_filter.yml (hunk 1 above)
Comment (filipecosta90) [reply tests/benchmarks/variable_length_filter.yml:23]: overall command rate in RPS

File: tests/benchmarks/variable_length_filter.yml (hunk 3 above)
Comment (filipecosta90) [reply tests/benchmarks/variable_length_filter.yml:4]: Yes, but not right away. We would need to add it to `defaults.yml` and I need to enable specifying the default remote env on the tool ( right now it only allows to inherit `kpis` and `exporter`.

**Pull request merged**